docs/storage/storage.md

```diff
@@ -0,0 +1,186 @@
+# Storage Design Notes
+## Status: (Very very)^N Draft
+### Author: Martin Smith
+
+## Tree Node Storage
+
+The node level of storage provides a fairly abstract tree model that is used to implement
+verifiable logs and maps. Most users will not need this level but it is important to know the
+concepts involved. The API for this is defined in `storage/tree_storage.go`. Related protos are in
+the `storage/storagepb` package.
+
+The model provides a versioned view of the tree. Revision numbers increase monotonically as the
+tree is mutated.
+
+### NodeIDs
+
+Nodes in storage are uniquely identified by a `NodeID`. This combines a tree path with
+a revision number. The path is effectively used as a bitwise subtree prefix in the tree. 
+
+The same `NodeID` objects are used by both logs and maps but they are interpreted differently.
+There are API functions that create them for each case. Mixing node ID types in API calls will
```

Comment:
It could but we don't currently do that. Users shouldn't need to create node ids. The underlying tree storage is shared between logs and maps and just works with the path bits.